pyquil/api/_qam.py

```diff
@@ -60,18 +61,36 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: Optional[int] = None,
```

Comment:
@karalekas Tommy had a question for you, following a suggestion you made on the first time this PR was made, that wasn't copied over to this new PR. https://github.com/rigetti/pyquil/pull/1079#discussion_r347106051